pkg/pillar/cmd/volumemgr/handlediskmetrics.go

```diff
@@ -132,9 +132,11 @@ func createOrUpdateDiskMetrics(ctx *volumemgrContext) {
 	log.Functionf("createOrUpdateDiskMetrics")
 	var diskMetricList []*types.DiskMetric
 	startPubTime := time.Now()
+	wdName := agentName + "metrics"
```

Comment:
I think the fix would be to mark the entire volumemanager as still running, not just a new "metrics" watchdog. With this change, just a new watchdog is introduced. And it's not related to the watchdog that could fire before. @milan-zededa, am I right here?

Reply:
@OhmSpectator the diskmetricstimertask is a separate top-level goroutine so it has its own wdName to detect if it somehow gets stuck.

Reply:
@eriknordmark, thanks, I got it!